crowbar_framework/app/models/proposal.rb

```diff
@@ -87,6 +92,27 @@ def raw_data=(value)
 
   private
 
+  # XXX: we need to be careful to not create an endless loop
+  def update_corresponding_proposal_object
+    proposal_object = ProposalObject.find_proposal_by_id(self.key)
+    if proposal_object
+      if proposal_object.raw_data != self.raw_data
+        increment_crowbar_revision! if proposal_object.crowbar_revision >= self.crowbar_revision
```

Comment:
> first, shouldn't we always increment the revision before saving (unless we do the save for a sync, of course)?
I'll go through the code again. I'm sure there is a reason for this, but I no longer remember ;)

> I'm not sure what the if condition is trying to protect against?

Which one?